packages/@aws-cdk/aws-dynamodb/lib/table.ts

```diff
@@ -48,6 +49,18 @@ export interface Attribute {
   readonly type: AttributeType;
 }
 
+export interface TableKMSOptions {
+  /**
+   * The principal that will have access to the KMS CMK via DynamoDB
+   */
+  readonly arnPrincipal: string;
```

Comment:
That looks wrong. You'd want to transparently grant permissions on the `IKey` when the `grantRead`, etc method get called on this table.

You might want to have a look at how S3 Bucket encryption is done, to have a feeling of what the API looks like, and how the `grant` methods operate there.